lib/src/git_backend.rs

```diff
@@ -537,6 +537,10 @@ impl Backend for GitBackend {
                 }
             }
         };
+        // Update the signatures to match the ones that were actually written to the object store
```

Comment:
You probably considered this already, but it might be possible to simplify the code above a little using the fact that we now own `contents` and can modify it directly. Then, you wouldn't need to re-calculate the timestamps below.

This can go in a separate PR/commit, of course. Or is there some reason it doesn't work?

Reply:
Yes, I did consider that but it wasn't obvious to me how to do it in a simpler way. We still need to recalculate the committer timestamp when we adjust it for collisions, right?

Reply:
I'm still thinking about this. Feel free to merge this before I try it; if I figure out some minor simplification I can always do another PR.

My minor annoyance is that we seem to be converting seconds to committer objects one too many times.

Reply:
Sure, I'll leave that for you to look into if you like. Thanks.